extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/ChaserThingHandler.java

```diff
@@ -0,0 +1,207 @@
+            }
+        }
+        return (values.size() > 0);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration configuration = getConfig();
+        if (configuration.get(CONFIG_DMX_ID) != null) {
+            channels.clear();
+            DmxBridgeHandler bridgeHandler = (DmxBridgeHandler) getBridge().getHandler();
+            List<BaseChannel> configChannels = BaseChannel.fromString((String) configuration.get(CONFIG_DMX_ID),
+                    bridgeHandler.getUniverseId());
+            logger.trace("found {} channels in {}", configChannels.size(), this.thing.getUID());
+            for (BaseChannel channel : configChannels) {
+                channels.add(bridgeHandler.getDmxChannel(channel, this.thing));
+            }
+            if (configuration.get(CONFIG_CHASER_STEPS) != null) {
+                if (parseChaserConfig((String) configuration.get(CONFIG_CHASER_STEPS))) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Chase configuration malformed in " + this.thing.getUID());
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Chase configuration missing in " + this.thing.getUID());
+            }
+            if (configuration.get(CONFIG_CHASER_RESUME_AFTER) != null) {
+                resumeAfter = (Boolean) configuration.get(CONFIG_CHASER_RESUME_AFTER);
+                logger.trace("set resumeAfter to {} in thing {}", resumeAfter, this.thing.getUID());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "DMX channel configuration missing in " + this.thing.getUID());
+        }
+
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo thingStatusInfo) {
+        if (thingStatusInfo.getStatus().equals(ThingStatus.OFFLINE)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.BRIDGE_OFFLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+            initialize();
+        }
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case CHANNEL_SWITCH:
+                for (Channel channel : channels) {
+                    channel.removeListener(channelUID);
+                }
+                break;
+            default:
+                logger.warn("channel {} not supported in thing {}", channelUID.getId(), this.thing.getUID());
+        }
+
+    }
+
+    @Override
+    public void updateState(ChannelUID channelUID, State state) {
+        if (channelUID.getId().equals(CHANNEL_SWITCH)) {
+            this.isRunning = (OnOffType) state;
```

Comment:
How should that happen? Switches only support OnOffType. But anyway: What do you suggest to catch that?